modules/standard/Sets.chpl

```diff
@@ -0,0 +1,496 @@
+    return new set(t, false);
+  }
+
+  /*
+    Return a new set that contains the difference of two sets.
+
+    :arg a: A set to take the difference of.
+    :arg b: A set to take the difference of.
+
+    :return: A new set containing the difference between `a` and `b`.
+    :rtype: `set(?t, ?)`
+  */
+  proc -(const ref a: set(?t, ?), const ref b: set(t, ?)): set(t, ?) {
+    return setDifference(a, b);
+  }
+
+  /*
+    Return a new set that contains the intersection of two sets.
+
+    :arg a: A set to take the intersection of.
+    :arg b: A set to take the intersection of.
+
+    :return: A new set containing the intersection of `a` and `b`.
+    :rtype: `set(?t, ?)`
+  */
+  proc setIntersection(const ref a: set(?t, ?),
+                     const ref b: set(t, ?)): set(t, ?) {
+    return new set(t, false);
+  }
+
+  /*
+    Return a new set that contains the intersection of two sets.
+
+    :arg a: A set to take the intersection of.
+    :arg b: A set to take the intersection of.
+
+    :return: A new set containing the intersection of `a` and `b`.
+    :rtype: `set(?t, ?)`
+  */
+  proc &(const ref a: set(?t, ?), const ref b: set(t, ?)): set(t, ?) {
+    return setIntersection(a, b);
+  }
+
+  /*
+    Return a new set that contains the symmetric difference of two sets.
+
+    :arg a: A set to take the symmetric difference of.
+    :arg b: A set to take the symmetric difference of.
+
+    :return: A new set containing the symmetric difference of `a` and `b`.
+    :rtype: `set(?t, ?)`
+  */
+  proc setSymmetricDiff(const ref a: set(?t, ?),
+                     const ref b: set(t, ?)): set(t, ?) {
+    return new set(t, false);
+  }
+
+  /*
+    Return the symmetric difference of two sets.
+
+    :arg a: A set to take the symmetric difference of.
+    :arg b: A set to take the symmetric difference of.
+
+    :return: A new set containing the symmetric difference of `a` and `b`.
+    :rtype: `set(?t, ?)`
+  */
+  proc ^(const ref a: set(?t, ?), const ref b: set(t, ?)): set(t, ?) {
+    return setSymmetricDiff(a, b);
+  }
+
+  /*
+    Return `true` if the sets `a` and `b` are equal. That is, they are the
+    same size and contain the same elements.
+
+    :arg a: A set to compare.
+    :arg b: A set to compare.
+
+    :return: `true` if two sets are equal.
+    :rtype: `bool`
+  */
+  proc ==(const ref a: set(?t, ?), const ref b: set(t, ?)): bool {
+    return false;
+  }
+
+  /*
+    Return `true` if the sets `a` and `b` are not equal.
+
+    :arg a: A set to compare.
+    :arg b: A set to compare.
+
+    :return: `true` if two sets are not equal.
+    :rtype: `bool`
+  */
+  proc !=(const ref a: set(?t, ?), const ref b: set(t, ?)): bool {
+    return false;
+  }
+
+  /*
+    Return `true` if `a` is a proper subset of `b`.
+
+    :arg a: A set to compare.
+    :arg b: A set to compare.
+
+    :return: `true` if `a` is a proper subset of `b`.
+    :rtype: `bool`
+  */
+  proc <(const ref a: set(?t, ?), const ref b: set(t, ?)): bool {
```

Comment:
If we have these comparison routines, why do we need methods like `isProperSubset` ?